waku/v2/peermanager/peer_connector.go

```diff
@@ -104,11 +103,20 @@ func (c *PeerConnectionStrategy) consumeSubscription(ctx context.Context, ch <-c
 		select {
 		case <-ctx.Done():
 			return
-		case p := <-ch:
-			select {
-			case <-ctx.Done():
-				return
-			case c.peerCh <- p:
+		default:
+			if !c.isPaused() {
+				select {
+				case <-ctx.Done():
+					return
+				case p, ok := <-ch:
+					if !ok {
```

Comment:
Do also take not of codeClimate complaining having deep nested conditionals. 
Code would become hard to maintain, it would be good to simplify this and break it down.